### ISSUE TITLE
Todo lists should not be present in the documentation

### ISSUE THREAD

**Issue by albert-github** [CGAL/cgal#6915]
In issue #6899 already a todo list was reported (self assigned in the mean time to @MaelRL), though there a far more packages where todo lists are present.

I think that todo lists should not be present in the documentation and that the mentioned items should either:
- be fixed
- create a new issue for it
- have already been fixed and the todo command  / text should be removed

The following projects have just "check generated documentation"
- Algebraic_foundations
- Algebraic_kernel_d
- Apollonius_graph_2
- Arrangement_on_surface_2
- Boolean_set_operations_2
- Bounding_volumes
- Convex_hull_d
- Envelope_2
- Envelope_3
- HalfedgeDS
- Inscribed_areas
- Jet_fitting_3
- Kernel_d
- Matrix_search
- Minkowski_sum_3
- Nef_3
- Partition_2
- Point_set_2
- Polyhedron
- Polynomial
- Polytope_distance_d
- Principal_component_analysis
- QP_solver
- Ridges_3
- SearchStructures
- Segment_Delaunay_graph_2
- Segment_Delaunay_graph_Linf_2
- Snap_rounding_2
- Surface_mesher
- Surface_sweep_2
- TDS_2
- TDS_3
- Voronoi_diagram_2

The following projects have multiple todo items:
- BGL
- Box_intersection_d
- Circular_kernel_2
- Convex_hull_3
- Hyperbolic_triangulation_2
- Kernel_23
- Mesh_3
- Number_types
- Periodic_4_hyperbolic_triangulation_2 (issue #6899)
- Polygon_mesh_processing
- Surface_mesh_approximation
- Surface_mesh_shortest_path
- Surface_mesh_skeletonization
- Tetrahedral_remeshing
- Three
- Triangulation_2

**Comment by afabri** [comment]
Note that the todo entries are NOT in the documentation at doc.cgal.org, but only in the documentations in the  testsuite.

**Comment by albert-github** [comment]
I agree that it is `not a bug` (as the end user doesn't see them unless one uses self generated documentation) and that it is a cleaning `issue`, though the question remains: how will it be possible to "push" the responsible person(s) for the package(s) to fix the problem?

**Comment by albert-github** [comment]
Even though the resulting todo  lists don't appear in the end documentation I think it would be better (quality of the documentation) to check whether to `\todo` commands are still relevant. A quick check shows that the following packages generate a `todo.html`:
```
Algebraic_foundations
Algebraic_kernel_d
Apollonius_graph_2
Arrangement_on_surface_2
BGL
Boolean_set_operations_2
Bounding_volumes
Box_intersection_d
Circular_kernel_2
Convex_hull_3
Convex_hull_d
Envelope_2
Envelope_3
HalfedgeDS
Hyperbolic_triangulation_2
Inscribed_areas
Jet_fitting_3
Kernel_23
Kernel_d
Matrix_search
Minkowski_sum_3
Nef_3
Number_types
Partition_2
Point_set_2
Polygon_mesh_processing
Polyhedron
Polynomial
Polytope_distance_d
Principal_component_analysis
QP_solver
Ridges_3
SearchStructures
Segment_Delaunay_graph_2
Segment_Delaunay_graph_Linf_2
Snap_rounding_2
Surface_mesher
Surface_mesh_approximation
Surface_mesh_shortest_path
Surface_mesh_skeletonization
Surface_sweep_2
TDS_2
TDS_3
Three
Triangulation_2
Voronoi_diagram_2
```

**Comment by albert-github** [comment]
I also see some `TODO` files:
```
Alpha_shapes_3/TODO
Apollonius_graph_2/TODO
Arrangement_on_surface_2/TODO
Arrangement_on_surface_2/test/Arrangement_on_surface_2/TODO
Cartesian_kernel/TODO
CGAL_ipelets/TODO
Circular_kernel_3/TODO
Filtered_kernel/TODO
GraphicsView/demo/Segment_Delaunay_graph_2/TODO
Hyperbolic_triangulation_2/TODO
Inscribed_areas/TODO
Interpolation/TODO
Lab/demo/Lab/TODO
Maintenance/deb/TODO
Mesh_2/TODO
Mesh_3/TODO
Nef_2/TODO
Nef_3/TODO
Nef_S2/TODO
Number_types/TODO
Segment_Delaunay_graph_2/TODO
Skin_surface_3/TODO
Spatial_searching/TODO
Straight_skeleton_2/TODO
Subdivision_method_3/TODO
Surface_mesher/TODO
Surface_mesh_simplification/TODO
Surface_mesh_skeletonization/TODO
Triangulation/TODO
Triangulation_2/TODO
Triangulation_3/TODO
Voronoi_diagram_2/TODO
```
a critical look here would not harm either.
